File: ProyectosCSharp/SoftProg/SoftProgPersistance/Almacen/DAO/ProductoDAO.cs
using SoftProgModel.Almacen;
using SoftProgModel.RRHH;
using SoftProgPersistance.DAO;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftProgPersistance.Almacen.DAO
{
    public interface ProductoDAO : IDAO<Producto>
    {
        BindingList<Producto> listarPorNombre(string nombre);
    }
}

File: Arquitectura_REST_SOAP/Backend/SoftProgDA/src/main/java/pe/edu/pucp/softprog/almacen/dao/ProductoDAO.java
package pe.edu.pucp.softprog.almacen.dao;

import java.util.ArrayList;
import pe.edu.pucp.softprog.almacen.model.Producto;
import pe.edu.pucp.softprog.dao.IDAO;

public interface ProductoDAO extends IDAO<Producto>{
    ArrayList<Producto> listarPorNombre(String nombre);
}
